fetcher.go

```diff
@@ -464,3 +474,18 @@ func getCacheStatus(isCacheHit bool) string {
 func subReqID(host, rsrc string) string {
 	return fmt.Sprintf("%x", crc32.ChecksumIEEE([]byte(host+rsrc)))
 }
+
+func createTraceparentFromTraceID(traceID string) string {
+	version := "00"
+	parentSpanID := generateRandomHex(8)
+	traceFlags := "01"
+
+	traceparent := fmt.Sprintf("%s-%s-%s-%s", version, traceID, parentSpanID, traceFlags)
+	return traceparent
+}
```

Comment:
instead of doing this manually, can we record this as a span?

* get in a context as you currently do (I would do this at the `Fetch` method in caboose.go)
  * use the parentSpanID from the `requestcontext`
  * turn it into a `spanContext` with [NewSpanContext](https://pkg.go.dev/go.opentelemetry.io/otel/trace#NewSpanContext)
  * Put that span context on the config used throughout caboose using [ContextWithRemoteSpanContext](https://pkg.go.dev/go.opentelemetry.io/otel/trace#ContextWithRemoteSpanContext)

Now you have a span context from upstream on the config for the overall fetch. on each subattempt that caboose makes in this `fetchResource` method you'll indicate that's a sub-span, using:
```
var span trace.Span
ctx, span = tracer.Start(ctx, "attempt")
defer span.End()
```

to get the current sub-span attached in to the request header of the http sub requests you'll do [this](https://github.com/oinume/go-http-client-trace-sample/blob/main/examples/otelhttp_client_trace/main.go#L42-L44)
```
clientTrace := otelhttptrace.NewClientTrace(ctx)
ctx = httptrace.WithClientTrace(ctx, clientTrace)
req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
```

Reply:
The id from `requestcontext` is supposed to be the traceID, no? Since its 32 hex chars. spanID is 16 hex chars. 

I still need a spanID to create the span context. I was hoping to get it from [this span](https://github.com/filecoin-saturn/caboose/blob/cb81e714c3eba70118509f2ef3b0180adb41ef8d/caboose.go#L262), but its spanID is all zeros 🤔 .

Reply:
* correct - `requestcontext` will give you the traceID.
* maybe just try https://pkg.go.dev/go.opentelemetry.io/otel/trace#SpanFromContext and get the span id from that - if there is one set in bifrost, you'll inherit it, and otherwise you'll be a child of an all 0 span, which should be okay